core/src/main/java/software/amazon/awssdk/core/SdkResponseMetadata.java

```diff
@@ -15,18 +15,23 @@
 
 package software.amazon.awssdk.core;
 
+import java.util.Collections;
 import java.util.Map;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkPublicApi;
 
 /**
- * Represents additional metadata included with a response from AWS. Response
+ * Represents additional metadata included with a response from a service. Response
  * metadata varies by service, but all services return an AWS request ID that
  * can be used in the event a service call isn't working as expected and you
  * need to work with AWS support to debug an issue.
  * <p>
  * Access to AWS request IDs is also available through the com.amazonaws.request
  * logger in the AWS SDK for Java.
  */
-public class ResponseMetadata {
+@Immutable
+@SdkPublicApi
+public class SdkResponseMetadata {
```

Comment:
Can we leave the metadata stuff out of it for now? I think we need some further design around that since there are some extra complexities with services like S3 that expose more modeled metadata.